File: src/internal/domain/entities/company_entity.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use crate::internal::domain::entities::user_entity::{serialize_uuid, deserialize_uuid};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct CompanyEntity {
    #[serde(serialize_with = "serialize_uuid", deserialize_with = "deserialize_uuid")]
    pub id: Uuid,
    #[serde(serialize_with = "serialize_uuid", deserialize_with = "deserialize_uuid")]
    pub user_id: Uuid,
    pub company_name: String,
    pub company_address: String,
    pub total_staff: u16,
}



impl CompanyEntity {
    pub fn new(id: Uuid, user_id: Uuid, company_name: String, company_address: String, total_staff: u16) -> Self {
        Self { id, user_id, company_name, company_address, total_staff }
    }
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ChangeCompanyEntity {
    #[serde(serialize_with = "serialize_uuid", deserialize_with = "deserialize_uuid")]
    pub user_id: Uuid,
    pub company_name: String,
    pub company_address: String,
    pub total_staff: u16,
}

impl ChangeCompanyEntity {
    pub fn new(user_id: Uuid, company_name: String, company_address: String, total_staff: u16) -> Self {
        Self { user_id, company_name, company_address, total_staff }
    }
}

File: internal/domain/entities/company_entity.go
package entities

import (
	"time"

	"github.com/gocql/gocql"
)

type CompanyEntity struct {
	Id             gocql.UUID `json:"id"`
	UserId         gocql.UUID `json:"userId"`
	CompanyName    string     `json:"companyName"`
	CompanyAddress string     `json:"companyAddress"`
	TotalStaff     int        `json:"totalStaff"`
	CreatedAt      time.Time  `json:"createdAt"`
	UpdatedAt      time.Time  `json:"updatedAt"`
}

type ChangeCompanyEntity struct {
	UserId         gocql.UUID `json:"userId"`
	CompanyName    string     `json:"companyName"`
	CompanyAddress string     `json:"companyAddress"`
	TotalStaff     int        `json:"totalStaff"`
	UpdatedAt      time.Time  `json:"updatedAt"`
}
